File: src/app/Class/Album.ts
export class Album {
  id: number;
  private albumName: string;
  private albumLengthSeconds: number;
  private releaseDate: Date;
  private company: string;
  private coverArtLink: string;
  private type: string;
  private trackCount: number;


  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.albumName = (obj && obj.albumName) || '';
    this.albumLengthSeconds = (obj && obj.albumLengthSeconds) || 0;
    this.releaseDate = (obj && obj.releaseDate) || null;
    this.company = (obj && obj.company) || '';
    this.coverArtLink = (obj && obj.coverArtLink) || '';
    this.type = (obj && obj.type) || '';
    this.trackCount = (obj && obj.trackCount) || 0;
  }

  // Getters

  public getId(): number {
    return this.id;
  }

  public getAlbumName(): string {
    return this.albumName;
  }

  public getAlbumLengthSeconds(): number {
    return this.albumLengthSeconds;
  }

  public getReleaseDate(): Date {
    return this.releaseDate;
  }

  public getCompany(): string {
    return this.company;
  }

  public getCoverArtLink(): string {
    return this.coverArtLink;
  }

  public getType(): string {
    return this.type;
  }

  public getTrackCount(): number {
    return this.trackCount;
  }

  // Setters

  public setAlbumName(albumName: string) {
    return (this.albumName = albumName);
  }

  public setAlbumLengthSeconds(albumLengthSeconds: number) {
    return (this.albumLengthSeconds = albumLengthSeconds);
  }

  public setReleaseDate(releaseDate: Date) {
    return (this.releaseDate = releaseDate);
  }

  public setCompany(company: string) {
    return (this.company = company);
  }

  public setCoverArtLink(coverArtLink: string) {
    return this.coverArtLink = coverArtLink;
  }

  public setType(type: string) {
    return this.type = type;
  }

  public setTrackCount(trackCount: number) {
    return this.trackCount = trackCount;
  }
}

File: src/main/java/musiccube/entities/Album.java
package musiccube.entities;

import com.fasterxml.jackson.annotation.JsonInclude;

import javax.persistence.*;
import java.util.Date;

@Entity
@JsonInclude(JsonInclude.Include.NON_NULL)
@Table(name = "album")
public class Album {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @Column
    private String albumName;

    @Column
    private int albumLengthSeconds;

    @Column
    private Date releaseDate;

    @Column
    private String company;

    @Column
    private String coverArtLink;

    @Column
    private String type;

    @Column
    private int trackCount;

    public Album() {}

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getAlbumName() {
        return albumName;
    }

    public void setAlbumName(String albumName) {
        this.albumName = albumName;
    }

    public int getAlbumLengthSeconds() {
        return albumLengthSeconds;
    }

    public void setAlbumLengthSeconds(int albumLengthSeconds) {
        this.albumLengthSeconds = albumLengthSeconds;
    }

    public Date getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(Date releaseDate) {
        this.releaseDate = releaseDate;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public String getCoverArtLink() {
        return coverArtLink;
    }

    public void setCoverArtLink(String coverArtLink) {
        this.coverArtLink = coverArtLink;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getTrackCount() {
        return trackCount;
    }

    public void setTrackCount(int trackCount) {
        this.trackCount = trackCount;
    }
}